insert_string_tpl.h

```diff
@@ -29,21 +29,13 @@
 #  define HASH_CALC_MASK HASH_MASK
 #endif
 #ifndef HASH_CALC_READ
-#  ifdef UNALIGNED_OK
-#    if BYTE_ORDER == LITTLE_ENDIAN
-#      define HASH_CALC_READ \
-          memcpy(&val, strstart, sizeof(val));
-#    else
-#      define HASH_CALC_READ \
-          memcpy(&val, strstart, sizeof(val)); \
-          val = ZSWAP32(val);
-#    endif
+#  if BYTE_ORDER == LITTLE_ENDIAN
+#    define HASH_CALC_READ \
+        memcpy(&val, strstart, sizeof(val));
 #  else
 #    define HASH_CALC_READ \
-        val  = ((uint32_t)(strstart[0])); \
-        val |= ((uint32_t)(strstart[1]) << 8); \
-        val |= ((uint32_t)(strstart[2]) << 16); \
-        val |= ((uint32_t)(strstart[3]) << 24);
+        memcpy(&val, strstart, sizeof(val)); \
+        val = ZSWAP32(val);
```

Comment:
Looks good, I wonder of GCC was missing this obvious byteswap idiom.  If so, this should actually end up being faster, as I think GCC will probably translate this to movbe on x86.